CONTRIBUTING.md

```diff
@@ -1,7 +1,82 @@
-* Contribute code via GitHub Pull Requests, see our [contribution page](https://community.openproject.org/projects/openproject/wiki/Contribution) for more information
-* See [Git Flow](https://community.openproject.org/projects/openproject/wiki/Git_Branching_Model)
+*Please also note that these rules should be acknowledged by everyone, but repository contributors might occasionally deviate from them for practical purposes, e.g. not fork the repo, but have a branch on the main repository.*
+
+## Contributors License Agreement
+
+External contributors have to sign a CLA before contributing to OpenProject.
+The [CLA can be found here](https://www.openproject.org/wp-content/uploads/2014/09/OPF-Contributor-License-Agreement_v.2.pdf) and has to be filled out and sent to cla@openproject.org. Additionally, a GPG signature has to be provided.
+
+## Development flow
+
+For contributing source code, please follow the Git Workflow below:
+
+- **Fork** OpenProject on GitHub
+- Clone your fork to your development machine: 
+
+```
+git clone git@github.com/<username>/openproject
+```
+
+- Optional: Add the original OpenProject repository as a remote, so you can fetch changes: 
+
+```
+git remote add upstream git@github.com:opf/openproject
+```
+
+- Make sure you're on the right branch. The main development branch is `dev`: 
+
+```
+git checkout dev
+```
+
+- Create a feature branch: 
+
+```
+git checkout -b feature/<short description of your feature>
+```
+
+- Make your changes, then push the branch into your ***own*** repository:
+
+```
+git push origin <your feature branch>
+```
+
+- Create a pull request against a branch of of the <opf/openproject> repository, containing a ***clear description*** of what the pull request attempts to change and/or fix. 
+
+If your pull request **does not contain a description** for what it does and what it's intentions are, we will reject it. If you are working on a specific work package from the [list](https://community.openproject.org/projects/openproject/work_packages?query_props=%7B%22c%22:%5B%22type%22,%22status%22,%22subject%22,%22assigned_to%22%5D,%22t%22:%22parent:desc%22,%22f%22:%5B%7B%22n%22:%22status_id%22,%22o%22:%22!%22,%22t%22:%22list_status%22,%22v%22:%5B%2217%22,%2223%22,%223%22,%2214%22,%226%22%5D%7D%5D,%22pa%22:1,%22pp%22:20%7D), you may include a link to that work package in the description, so we can track your work.
+
+We will then review your pull request. Please note that you can add commits after the pull request has been created by pushing to the branch in your fork.
+
+## Important notes
+
+To ensure a smooth workflow for, please take note of the following:
+
+### Testing
+
+Please add tests to your code to verify functionality, especially if it is a new feature.
+
+Pull requests will be verified by TravisCI as well, but please run them locally as well. We have a lot of pull requests coming in and it takes some time to run the complete suite for each one.
```

Comment:
Maybe we should say here that it would be best if the tests were green when run locally. So there is a chance that Travis needs to test this PR only once.